src/openms/source/ANALYSIS/ID/FragmentIndex.cpp

```diff
@@ -119,64 +116,85 @@ namespace OpenMS
                 continue;
               }
               #pragma omp critical (FIIndex)
-              {  
-                fi_peptides_.push_back({modified_peptide, protein_idx, modified_mz});
-              } 
+              {
+                fi_peptides_.push_back({static_cast<UInt32>(i), modification_idx, digested_peptide, modified_mz});
+                ++modification_idx;
+              }
             }
           }
           else
           {
-            if (peptide_min_mass_ < unmodified_mz && unmodified_mz < peptide_max_mass_) // TODO: <= instad of <?
+            if (peptide_min_mass_ <= unmodified_mz && unmodified_mz <= peptide_max_mass_)
             {
               #pragma omp critical (FIIndex)
-              {  
-                fi_peptides_.push_back({unmod_peptide, protein_idx, unmodified_mz});
+              {
+                fi_peptides_.push_back({static_cast<UInt32>(i), 0, digested_peptide, unmodified_mz});
               }
-            }              
+            }
           }
         }
-        #pragma omp atomic
-        protein_idx++;
+        //#pragma omp atomic
+        //protein_idx++;
       }
       if (skipped_peptides > 0)
       {
         OPENMS_LOG_WARN << skipped_peptides << " peptides skipped due to unkown AA \n";
       }        
       // sort the peptide vector, critical for following steps
-      sort(fi_peptides_.begin(), fi_peptides_.end(), [](const Peptide& a, const Peptide& b){ return a.precursor_mz < b.precursor_mz; });
+      sort(fi_peptides_.begin(), fi_peptides_.end(), [](const sPeptide& a, const sPeptide& b){ return a.precursor_mz < b.precursor_mz; });
   }
 
   void FragmentIndex::build(const std::vector<FASTAFile::FASTAEntry>& fasta_entries)
   {
       // get the spectrum generator and set the ion-types
       TheoreticalSpectrumGenerator tsg;
       auto tsg_params = tsg.getParameters();
-      tsg_params.setValue("add_a_ions", add_a_ions_);
-      tsg_params.setValue("add_b_ions", add_b_ions_);
-      tsg_params.setValue("add_c_ions", add_c_ions_);
-      tsg_params.setValue("add_x_ions", add_x_ions_);
-      tsg_params.setValue("add_y_ions", add_y_ions_);
-      tsg_params.setValue("add_z_ions", add_z_ions_);
-      tsg.setParameters(tsg_params); // TODO: this was missing before!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+      auto this_params = getParameters();
+      tsg_params.setValue("add_a_ions", this_params.getValue("add_a_ions"));
```

Comment:
Could be done based on activation method instead.
Note: z-ions are currently wrong

Reply:
hmm good idea but I think one doesn't want to build the fragment index on the a-ions for HCD data. Maybe this part should stay user customizable?